### PR TITLE
New struct monster_summary documentation

### PR DESCRIPTION
**Pull request by Jawshoeuh**
Added the struct mentioned in https://github.com/UsernameFodder/pmdsky-debug/issues/164

Taking better naming conventions for the functions because I'm uncertain if they're good.

### CODE REVIEW

File: headers/types/dungeon_mode/dungeon_mode_common.h [hunk 1]
```diff
@@ -62,4 +62,61 @@ struct move {
...
+    struct tactic_id_8 tactic;  // 0x54
+    undefined field_0x55;
+    undefined field_0x56;
+    undefined field_0x57;
+    // 0x58: Appears to be a list of all the currently inflicted statues in their enum form. The
+    // last entry (30th) appears to always be STATUS_NONE to serve as a terminator for the list.
+    // While in ground mode, it's always filled with STATUS_NONE.
+    struct status_id_8 active_statuses[30];
+};
+ASSERT_SIZE(struct monster_summary, 118);
```
Comment (UsernameFodder) [inline headers/types/dungeon_mode/dungeon_mode_common.h:120]: As is there's two bytes of implicit padding at the end of the struct. Do you have confirmation (e.g., from a `MemAlloc` call or from the way they're laid out contiguously in arrays) that the struct size really is 118, as opposed to 120?

**Review by UsernameFodder — COMMENTED** [review]
Note: I fixed some formatting issues, and defined `struct status_id_8`, and moved `struct monster_summary` to `dungeon_mode_common.h` where I think it fits better (and has access to `status_id_8`). The only remaining issue is with the sizing of `struct monster_summary`.

File: headers/types/dungeon_mode/dungeon_mode_common.h (hunk 1 above)
Comment (Jawshoeuh) [reply headers/types/dungeon_mode/dungeon_mode_common.h:120]: You are correct. The function that call CreateMonsterSummaryFromMonster uses 0x78 (120) on the stack.

That size also perfectly lines up with the empty space on the dungeon struct where the struct exists. (Seems like I forgot to update the dungeon struct. I'll add it on there as well.)

**Review by UsernameFodder — APPROVED** [review]
Thanks!

**Pull request merged**